src/series/bar.js

```diff
@@ -18,30 +18,41 @@
                 var container = d3.select(this);
                 var g = fc.utilities.simpleDataJoin(container, 'bar', data, xValue);
 
-                // enter
-                g.enter()
-                    .append('path');
-
                 var width = barWidth(data.map(xValueScaled));
 
                 var pathGenerator = fc.svg.bar()
                     .width(width);
 
-                g.each(function(d, i) {
+                var x = function(d, i) { return xValueScaled(d, i); },
+                    y0 = function(d, i) { return y0Value(d, i); },
+                    barTop = function(d, i) { return yScale(y0(d, i) + y1Value(d, i)); },
+                    barBottom = function(d, i) { return yScale(y0(d, i)); };
 
-                    var x = xValueScaled(d, i),
-                        y0 = y0Value(d, i),
-                        barBottom = yScale(y0),
-                        barTop = yScale(y0 + y1Value(d, i));
+                g.enter()
+                    .each(function(d, i) {
+                        d3.select(this)
+                            .attr('transform', 'translate(' + x(d, i) + ', ' + barBottom(d, i) + ')');
+                    })
+                    .append('path')
+                    .each(function(d, i) {
+                        pathGenerator.x(0)
+                            .y(0)
+                            .height(0);
+
+                        d3.select(this)
+                            .attr('d', pathGenerator([d]));
+                    });
 
-                    var g = d3.select(this)
-                        .attr('transform', 'translate(' + x + ', ' + barTop + ')');
+                g.each(function(d, i) {
+                    pathGenerator.x(0)
+                        .y(0)
+                        .height(function() { return barBottom(d, i) - barTop(d, i); });
 
-                    pathGenerator.x(d3.functor(0))
-                        .y(d3.functor(0))
-                        .height(function() { return barBottom - barTop; });
+                    var barGroup = d3.select(this);
 
-                    g.select('path')
+                    d3.transition(barGroup)
+                        .attr('transform', 'translate(' + x(d, i) + ', ' + barTop(d, i) + ')')
```

Comment:
That's a good point. Translate it is!

On Tue, 26 May 2015 21:53 Colin Eberhardt <notifications@github.com> wrote:

> In src/series/bar.js
> <https://github.com/ScottLogic/d3-financial-components/pull/376#discussion_r31078621>
> :
>
> >
> > -                    g.select('path')
> > +                    d3.transition(barGroup)
> > +                        .attr('transform', 'translate(' + x(d, i) + ', ' + barTop(d, i) + ')')
>
> One disadvantage of a scale transform is that it would effect any stroke
> applied to the bar. Doesn't sound viable to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ScottLogic/d3-financial-components/pull/376/files#r31078621>
> .
>